cluster-api/src/main/java/io/aiven/klaw/clusterapi/config/JwtRequestFilter.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@Slf4j
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+  public static final String USERNAME = "username";
+  public static final String TOKEN = "token";
+  private final JwtTokenUtilService jwtTokenUtil;
+  private final UserDetailsService userDetailsService;
+
+  public JwtRequestFilter(
+      UserDetailsService userDetailsService, JwtTokenUtilService jwtTokenUtilService) {
+    this.userDetailsService = userDetailsService;
+    this.jwtTokenUtil = jwtTokenUtilService;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+      throws ServletException, IOException {
+    final Map<String, String> userTokenMap = new HashMap<>();
+    if (isUnAuthorizedRequest(request, response, userTokenMap)) return;
+
+    validateToken(request, response, chain, userTokenMap);
+  }
+
+  private boolean isUnAuthorizedRequest(
+      HttpServletRequest request, HttpServletResponse response, Map<String, String> userTokenMap)
+      throws IOException {
+    final String requestTokenHeader = request.getHeader("Authorization");
+    String jwtToken;
+    final String bearer = "Bearer ";
+    // JWT Token is in the form "Bearer token". Remove Bearer word and get
+    // only the Token
+    if (requestTokenHeader != null && requestTokenHeader.startsWith(bearer)) {
+      jwtToken = requestTokenHeader.substring(bearer.length());
+      userTokenMap.put(TOKEN, jwtToken);
+      return extractUsernameFromToken(response, userTokenMap, jwtToken);
+    } else {
+      if (requestTokenHeader != null) {
+        log.warn(
+            "Invalid JWT : {}",
+            requestTokenHeader.substring(0, Math.min(requestTokenHeader.length(), 10)));
+      }
+
+      log.warn("JWT Token does not begin with Bearer String");
+    }
+    return false;
+  }
+
+  private boolean extractUsernameFromToken(
+      HttpServletResponse response, Map<String, String> userTokenMap, String jwtToken)
+      throws IOException {
+    String username;
+    try {
+      username = jwtTokenUtil.getUsernameFromToken(jwtToken);
+      userTokenMap.put(USERNAME, username);
+    } catch (IllegalArgumentException e) {
+      log.info("Unable to get JWT Token ", e);
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+      return true;
+    } catch (ExpiredJwtException e) {
+      log.info("JWT Token has expired", e);
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+      return true;
+    } catch (Exception e) {
+      log.info("Token validation errors ", e);
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
+      return true;
+    }
+    return false;
+  }
```

Comment:
```suggestion
    try {
      username = jwtTokenUtil.getUsernameFromToken(jwtToken);
      userTokenMap.put(USERNAME, username);
      return false;
    } catch (IllegalArgumentException e) {
      log.info("Unable to get JWT Token ", e);
    } catch (ExpiredJwtException e) {
      log.info("JWT Token has expired", e);
    } catch (Exception e) {
      log.info("Token validation errors ", e);
    }
    response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
    return true;
  }
```
This way could decrease number of lines